app/src/androidTest/java/chat/rocket/android/members/ui/MembersFragmentTest.kt

```diff
@@ -34,10 +34,10 @@ class MembersFragmentTest {
     @Before
     fun setUp() {
         try {
-            login_if_user_is_logged_out()
-            navigate_to_channel_details()
+            loginIfUserIsLoggedOut()
```

Comment:
Are you counting on this to throw an exception if the user is already logged in?  Is that safe?  Is it guaranteed that no other exception could be thrown?  I just wonder if there is a more positive test whether the user is logged in...

Reply:
No the logic here is that if the user is already logged in then there will be a NoMatchingViewException for which we have to do nothing just proceed further in this case navigateToChannelDetails but if the user is not logged in then we have to make the user login first then navigate further

Reply:
I have to check this logic in some tests because the tests should be independent of each other (best practices) Also it saves tests from failing, there can be a case in which a test which tests a scenario after user authentication is performed before its actual login in the app. So in that case it will perform login first and then test the required thing